src/boosting/goss.hpp

```diff
@@ -97,18 +70,23 @@ class GOSS: public GBDT {
       is_use_subset_ = true;
     }
     // flag to not bagging first
-    bag_data_cnt_ = num_data_;
+    bag_data_cnt_ = num_data_; 
   }
 
-  data_size_t BaggingHelper(data_size_t start, data_size_t cnt, data_size_t* buffer) override {
+  void ResetConfig(const Config* config, bool is_change_dataset, 
+          std::vector<score_t, Common::AlignmentAllocator<score_t, kAlignedSize>>& gradients, 
+          std::vector<score_t, Common::AlignmentAllocator<score_t, kAlignedSize>>& hessians) override {}
+
+ protected:
+  data_size_t Helper(data_size_t start, data_size_t cnt, data_size_t* buffer, score_t* gradients, score_t* hessians) {
     if (cnt <= 0) {
       return 0;
     }
     std::vector<score_t> tmp_gradients(cnt, 0.0f);
     for (data_size_t i = 0; i < cnt; ++i) {
       for (int cur_tree_id = 0; cur_tree_id < num_tree_per_iteration_; ++cur_tree_id) {
         size_t idx = static_cast<size_t>(cur_tree_id) * num_data_ + start + i;
-        tmp_gradients[i] += std::fabs(gradients_[idx] * hessians_[idx]);
+        tmp_gradients[i] += std::fabs(gradients[idx] * hessians[idx]);
```

Comment:
>gradients[idx] * hessians[idx [](http://example.com/codeflow?start=38&length=29)

you are changing a member variant to a parameter pointer. although the change is so small, but you may add more code for that because you don't know what the caller will pass to this function, so you should check if it is the null pointer or not.

Reply:
Same as https://github.com/microsoft/LightGBM/pull/4827#discussion_r830434879